pkg/chunk/table_manager.go

```diff
@@ -202,36 +207,65 @@ func (m *TableManager) bucketRetentionLoop() {
 // SyncTables will calculate the tables expected to exist, create those that do
 // not and update those that need it.  It is exposed for testing.
 func (m *TableManager) SyncTables(ctx context.Context) error {
-	expected := m.calculateExpectedTables()
-	level.Info(util.Logger).Log("msg", "synching tables", "num_expected_tables", len(expected), "expected_tables", len(expected))
+	for i, client := range m.clients {
+		if client == nil {
+			continue
+		}
+		lastClient := i == len(m.clients)-1
+		expected := m.calculateExpectedTables(i)
+		if lastClient {
+			level.Info(util.Logger).Log("msg", "synching tables", "num_expected_tables", len(expected), "expected_tables", len(expected))
+		} else {
+			level.Info(util.Logger).Log("msg", "synching tables [old config]", "num_expected_tables", len(expected), "expected_tables", len(expected))
+		}
 
-	toCreate, toCheckThroughput, toDelete, err := m.partitionTables(ctx, expected)
-	if err != nil {
-		return err
-	}
+		toCreate, toCheckThroughput, toDelete, err := m.partitionTables(ctx, client, expected)
+		if err != nil {
+			return err
+		}
 
-	if err := m.deleteTables(ctx, toDelete); err != nil {
-		return err
-	}
+		if err := m.deleteTables(ctx, client, toDelete); err != nil {
+			if lastClient {
+				return err
+			}
+			level.Error(util.Logger).Log("msg", "error in deleting tables from TableClient")
```

Comment:
This log output states there was an error, but offers no context or the error itself. Can you add more details to this line in order to make it useful for troubleshooting?

Reply:
👍 It would be good to print the error in this log statement